app/javascript/components/public_health/query/JurisdictionFilter.js

```diff
@@ -20,13 +21,26 @@ class JurisdictionFilter extends React.Component {
     };
   }
 
-  handleJurisdictionChange = jurisdiction_path => {
-    this.setState({ jurisdiction_path }, () => {
-      const jurisdiction = Object.keys(this.props.jurisdiction_paths).find(id => this.props.jurisdiction_paths[parseInt(id)] === jurisdiction_path);
-      if (jurisdiction) {
+  // This update is necessary since sticky settings are loaded after this component is constructed
+  componentDidUpdate() {
+    if (this.state.jurisdiction_path !== this.props.jurisdiction_paths[this.props.jurisdiction]) {
+      this.setState({ jurisdiction_path: this.props.jurisdiction_paths[this.props.jurisdiction] });
+      this.setState({ jurisdiction_input: this.props.jurisdiction_paths[this.props.jurisdiction] });
```

Comment:
is there a reason we're tracking both `jurisdiction_path` and `jurisdiction input`? seems like they're always the same so you could used `jurisdiction_path` as the value in the select

Reply:
I think it's because of the comment on line 32

Reply:
Is there a reason we can't move the conditional on line 39 inside the setState callback like it was done originally?  Seems redundant to track it twice

Reply:
After some digging, the componentDidUpdate and second state variable are needed, but I'd rather track `jurisdiction_id` instead of `jurisdiction_input` and use the id in the conditional in the the component did update

Reply:
[3ab26e5](https://github.com/SaraAlert/SaraAlert/pull/1231/commits/3ab26e539d45764e6c304dc7125f93b5fd81c518)